File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/item/InventoryMutation.java
package top.likoslupus.cellulosesz.api.item;

import top.likoslupus.cellulosesz.api.platform.operation.PlatformResult;

/**
 * Prepared exact-slot inventory mutation with explicit conflict and rollback outcomes.
 */
public interface InventoryMutation {

    PlatformResult<Void> commit();

    PlatformResult<Void> rollback();

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/item/InventoryMutation.kt
package top.likoslupus.cellulosesz.api.item

import top.likoslupus.cellulosesz.api.platform.operation.PlatformResult

/** Prepared exact-slot inventory mutation with explicit conflict and rollback outcomes. */
public interface InventoryMutation {

    public fun commit(): PlatformResult<Void>

    public fun rollback(): PlatformResult<Void>

}
